src/pages/[id].page.jsx

```diff
@@ -32,17 +32,26 @@ export default function Details({ details }) {
 
   // dynamic imports need to be asynchronous
   useEffect(() => {
-    const updateCodeBlocks = async () => {
-      const html = await import(`../code-examples/${slug}/html.js`);
-      setHtmlCode(html.default);
+    const updateSandpackExamples = async () => {
+      try {
+        const html = await import(`../code-examples/${slug}/html.js`);
+        setHtmlCode(html.default);
 
-      const css = await import(`../code-examples/${slug}/css.js`);
-      setCssCode(css.default);
+        const css = await import(`../code-examples/${slug}/css.js`);
+        setCssCode(css.default);
 
-      const javascript = await import(`../code-examples/${slug}/javascript.js`);
-      setJavascriptCode(javascript.default);
+        const javascript = await import(
+          `../code-examples/${slug}/javascript.js`
+        );
+        setJavascriptCode(javascript.default);
+      } catch (e) {
+        if (htmlCode === '' || cssCode === '' || javascriptCode === '') {
+          const sandpackContainer = document.querySelector('.sandpack');
+          sandpackContainer.style.display = 'none';
```

Comment:
Using document selectors can be an anti-pattern for react-based projects.

How would you feel about creating a 'state' value for the sandPack:
```
const [canRenderSandpack, setCanRenderSandpack] = useState(true)
```
- then here on line ```49 and 50``` => ```setCanRenderSandpack(false)```
- then in the render:

{canRenderSandpack ? <figure>
            <Sandpack
              template="vanilla"
              theme={levelUp}
              files={{
                '/index.html': {
                  code: htmlCode,
                  active: true,
                },
                'index.css': cssCode,
                '/index.js': javascriptCode,
                '/src/index.js': {
                  code: 'import "../index.css"; import "../index.js"',
                  hidden: true,
                },
              }}
              options={{
                showLineNumbers: true,
                wrapContent: true,
                editorHeight: 480,
              }}
            />
          </figure>
          <figcaption>
            Once focused on the code editor:
            <p>
              Press
              {' '}
              <kbd>Enter</kbd>
              {' '}
              to enter the code editor and change the
              code.
            </p>
            <p>
              Press
              {' '}
              <kbd>Escape</kbd>
              {' '}
              to leave the code editor.
            </p>
          </figcaption> : null}

Reply:
☝️ This would be a good approach. I would point out that we should conditionally render `<article className="sandpack">`, not just the `Sandpack` component. I would maybe make the default state `false` for `canRenderSandpack`, then only set it to `true` at the end of the dynamic imports. That way it only renders if we succeed, rather than optimistically rendering nothing, then hiding it once things fail.

Also I don't think we need the `if` statement to check whether any of the samples has a value. If we're in the `catch` block, we know something went wrong and should proceed to not render the Sandpack stuff.

Reply:
I was thinking about what could go in the ```catch``` and maybe we could do something like:
```
canRenderSandpack && setCanRenderSandpack(false)
```

Reply:
Phew-ok....let me make sure I understand what we're doing (this is just for me, trying to re-explain what you both have suggested 🤯)

I'm storing state in new variables, `canRenderSandpack` and `setCanRenderSandpack`. The default state is set to false, just in case we can't get the code loaded from the dynamic imports. In the try block, I set `setCandRenderSandpack` to true after setting all of the code languages (roughly line 48). Then the catch block...also has `setCandRenderSandpack(false)`? 

At line 65-ish, is the JSX expression, that's checking the state of `canRenderSandpack`. If it's true, we render all of the sandpack stuff, article, h2 and figcaption included. 

Is that how this code is being read? It seems to be working the same, I just need to make sure I'm reading and understanding it. https://github.com/sparkbox/accessible-components/pull/132/commits/eee2c42fc89b8e2b0fd5680af469d84d679a8c00

Reply:
Yep! Nailed it!